### ISSUE TITLE
[Enhancement]: Add "out of stock" and "backorder" products to sample_products.csv

### ISSUE THREAD

**Issue by nielslange** [woocommerce/woocommerce#41675]
### Describe the solution you'd like

At [WooCommerce Blocks](https://github.com/woocommerce/woocommerce-blocks), we’re currently [skipping one E2E test](https://github.com/woocommerce/woocommerce-blocks/blob/434c0e3f453f35042b1df711e6fc139addb76124/tests/e2e/tests/product-collection/product-collection.block_theme.spec.ts#L206-L208) which checks the filter for products based on their stock status. Our E2E tests rely on the sample products from [`sample_products.csv`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/sample-data/sample_products.csv). 

To test against both "out of stock" and "backorder" products, I propose adding two additional products, which these stock statuses, to [`sample_products.csv`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/sample-data/sample_products.csv). 

cc: @imanish003 

### Describe alternatives you've considered

For the particular issue that [WooCommerce Blocks](https://github.com/woocommerce/woocommerce-blocks) is facing, I considered the following alternatives:
1. Change the stock status of two of the existing products in `sample_products.csv` to "out of stock" and "backorder". → This approach might interfere with existing E2E tests of other developing teams.
2. Import two additional products, with the expected stock status, while spinning up the E2E environment in [WooCommerce Blocks](https://github.com/woocommerce/woocommerce-blocks). → Only [WooCommerce Blocks](https://github.com/woocommerce/woocommerce-blocks) would benefit from such an approach, while other development teams might also have an interest in running tests against the product stock status.
3. Before running the corresponding test case in [WooCommerce Blocks](https://github.com/woocommerce/woocommerce-blocks), adjust the stock status of one of the initially imported products accordingly and revert that stock status change after the test run. → This will increase the overall running time of the test suits.

**Comment by imanish003** [comment]
In case you find it useful, there was a related discussion on GitHub which you can check out at: https://github.com/woocommerce/woocommerce-blocks/discussions/10099.

**Comment by nielslange** [comment]
Thanks for pointing out that discussion, @imanish003.

**Comment by vladolaru** [comment]
Hey @nielslange — I checked current trunk and this is still not implemented as a sample-data change, although the original Product Collection E2E blocker has been solved another way.

Current state:

- `sample_products.csv` still has 25 rows, all `In stock?=1`, blank `Stock`, and `Backorders allowed?=0`.
- `sample_products.xml` also has all product/product-variation items as `_stock_status=instock`, blank `_stock`, and `_backorders=no`.
- Current Blocks E2E setup imports `sample_products.xml`, not CSV, and #43548 unskipped the Product Collection stock-status test by marking `T-Shirt with Logo` out of stock during E2E setup.

So this issue is no longer needed to unblock that specific test. The linked test-data discussion also points in the same direction: the historical Product Collection tests were working from shared XML-imported fixture data, not the sample CSV alone. The issue is still valid, though, if we want reusable sample data that includes unavailable products for broader development/testing.

There is also open PR #58533 refreshing `sample_products.csv`. As of this check, the current branch has 59 rows and adds one out-of-stock `Socks` row, but I did not see a row with `In stock?=backorder` / `onbackorder`; `Backorders allowed?` remains `0` for every row, and the PR does not update `sample_products.xml`. That PR is also carrying broader product/image/grouped-product decisions, so I would suggest coordinating there rather than opening a separate competing sample-data update.

Before implementation, I think we need to decide:

- should merchant-facing sample data intentionally include unavailable products;
- should CSV and XML stay semantically aligned;
- should the backorder fixture use `In stock?=backorder` only, or also set `Backorders allowed?` to exercise managed-stock backorder behavior.

My recommendation is to keep this open only if we want that reusable fixture coverage. If the only goal was the historical Product Collection E2E blocker, this can be closed as superseded by #42284 / #43548.

**Comment by flakas** [comment]
> should merchant-facing sample data intentionally include unavailable products;

@vladolaru looks like the primary intent of the ticket is developer tooling, so I would suggest to not change anything that might affect merchant-facing flows (e.g. onboarding) without @timmy5685's input.

And as Niels' ask was an E2E improvement and that's now handled - can we close this as superseded by #43548 / #42284? If we later decide we want reusable unavailable-product data for broader dev/testing, we can pick that up separately as its own (test-scoped) fixture.

**Comment by oaratovskyi** [comment]
Thanks @nielslange for the original write-up, and @vladolaru and @flakas for the recent check-in. I took another look, and I agree we can close this as superseded.

- #43548 resolved the original E2E blocker by setting `T-Shirt with Logo` out of stock in the test setup, and #42284 closed with it.
- The discussion in #44957 favored dedicated fixture data for Playwright E2E rather than shared sample data.
- The automated paths involved here do not consume `sample_products.csv`: onboarding uses `experimental_fashion_sample_9_products.csv`, while Blocks E2E and performance tests use the XML. The CSV remains available for manual import, so broader changes to that sample set belong in #58533, which is already refreshing it and includes an out-of-stock product.

One separate test gap remains: the Product Collection stock-status test mentions all three statuses but only covers `Out of stock`. If we want that coverage, we can open a focused follow-up to set a visible product such as `Long Sleeve Tee` to `onbackorder` with `wp eval`, then assert `On backorder` in the existing test. The importer, Product Collection block, Store API, and REST API already support the status, so no production-code change is needed.

We can also ask on #58533 whether the refreshed CSV should include one backorder product.

Closing as superseded by #43548 and #42284. If broader sample-data coverage is still wanted, let’s continue that in #58533.
